tools/xdp/s2n-quic-xdp/src/mmap.rs

```diff
@@ -19,6 +19,12 @@ pub struct Mmap {
     len: usize,
 }
 
+/// Safety: Mmap pointer can be sent between threads
+unsafe impl Send for Mmap {}
+
+/// Safety: Mmap pointer can be shared between threads
+unsafe impl Sync for Mmap {}
```

Comment:
Can an Mmap shared between threads be safely defered? I'm not convinced by the comment. Maybe I'm missing something.

Reply:
I assume it is it safe because there is no mutable access to the backing memory.

Reply:
`Sync` doesn't say anything about mutability. Linking the docs here: https://doc.rust-lang.org/std/marker/trait.Sync.html

> The precise definition is: a type T is [Sync](https://doc.rust-lang.org/std/marker/trait.Sync.html) if and only if &T is [Send](https://doc.rust-lang.org/std/marker/trait.Send.html). In other words, if there is no possibility of [undefined behavior](https://doc.rust-lang.org/reference/behavior-considered-undefined.html) (including data races) when passing &T references between threads.